File: php/Lasagna.php
<?php

declare(strict_types=1);

class Lasagna
{
    function expectedCookTime()
    {
        return 40;
    } 

    function remainingCookTime($elapsed_minutes)
    {
        $time = $this->expectedCookTime();

        return $time - $elapsed_minutes;
    }

    function totalPreparationTime($layers_to_prep)
    {
        return $layers_to_prep * 2;
    }

    function totalElapsedTime($layers_to_prep, $elapsed_minutes)
    {
        $layers = $this->totalPreparationTime($layers_to_prep);

        return $elapsed_minutes + $layers;
    }

    function alarm()
    {
        return "Ding!";
    }
}

File: elixir/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven do
    40
  end

  def remaining_minutes_in_oven(time) do
    expected_minutes_in_oven() - time
  end

  def preparation_time_in_minutes(time) do
    time * 2
  end

  def total_time_in_minutes(layers, minutes)  do
        layers * 2 + minutes
  end

  def alarm do
    "Ding!"
  end
end
